ps_wirepayment.php

```diff
@@ -211,54 +211,36 @@ public function hookDisplayPaymentReturn($params)
             return;
         }
 
-        $state = $params['order']->getCurrentState();
-        if (
-            in_array(
-                $state,
-                [
-                    Configuration::get('PS_OS_BANKWIRE'),
-                    Configuration::get('PS_OS_OUTOFSTOCK'),
-                    Configuration::get('PS_OS_OUTOFSTOCK_UNPAID'),
-                ]
-        )) {
-            $bankwireOwner = $this->owner;
-            if (!$bankwireOwner) {
-                $bankwireOwner = '___________';
-            }
-
-            $bankwireDetails = Tools::nl2br($this->details);
-            if (!$bankwireDetails) {
-                $bankwireDetails = '___________';
-            }
+        $bankwireOwner = $this->owner;
+        if (!$bankwireOwner) {
+            $bankwireOwner = '___________';
+        }
 
-            $bankwireAddress = Tools::nl2br($this->address);
-            if (!$bankwireAddress) {
-                $bankwireAddress = '___________';
-            }
+        $bankwireDetails = Tools::nl2br($this->details);
+        if (!$bankwireDetails) {
+            $bankwireDetails = '___________';
+        }
 
-            $totalToPaid = $params['order']->getOrdersTotalPaid() - $params['order']->getTotalPaid();
-            $this->smarty->assign([
-                'shop_name' => $this->context->shop->name,
-                'total' => $this->context->getCurrentLocale()->formatPrice(
-                    $totalToPaid,
-                    (new Currency($params['order']->id_currency))->iso_code
-                ),
-                'bankwireDetails' => $bankwireDetails,
-                'bankwireAddress' => $bankwireAddress,
-                'bankwireOwner' => $bankwireOwner,
-                'status' => 'ok',
-                'reference' => $params['order']->reference,
-                'contact_url' => $this->context->link->getPageLink('contact', true),
-            ]);
-        } else {
-            $this->smarty->assign(
-                [
-                    'status' => 'failed',
-                    'contact_url' => $this->context->link->getPageLink('contact', true),
-                ]
-            );
+        $bankwireAddress = Tools::nl2br($this->address);
+        if (!$bankwireAddress) {
+            $bankwireAddress = '___________';
         }
 
+        $totalToPaid = $params['order']->getOrdersTotalPaid() - $params['order']->getTotalPaid();
```

Comment:
Operation with float values is not 100% accurate. Safer to use a dedicated library for this purpose.
But here I suppose the rounding issues are hidden by price formatting down below.